src/table.js

```diff
@@ -1322,6 +1322,72 @@ Table.prototype.sampleRowKeysStream = function(gaxOptions) {
   ]);
 };
 
+/**
+ * Returns true in callback if consitency check is true for
+ * generated consistency token.
+ * In-case consistency check returns false, retrial is done in interval
+ * of 5 seconds till 10 minutes, after that it returns false.
+ *
+ * @param {function(?Error, ?boolean)} [callback]
+ * The function which will be called with the result of the API call.
+ */
+Table.prototype.waitForReplication = function(callback) {
+  let reqOpts = {
+    name: this.id,
+  };
+
+  this.bigtable
+    .request({
+      client: 'BigtableTableAdminClient',
+      method: 'generateConsistencyToken',
+      reqOpts: reqOpts,
+    })
+    .then(resp => {
+      const cToken = resp[0].consistencyToken;
+
+      if (checkConsistency(cToken)) {
+        return callback(null, true);
+      } else {
+        // check consistency in every 5 seconds
+        let intervalCall = setInterval(
+          cToken => {
+            if (checkConsistency(cToken)) {
+              clearInterval(intervalCall);
+              clearTimeout(timeOutCall);
+              callback(null, true);
+            }
+          },
+          5 * 1000,
+          cToken
+        );
+
+        // after 10 minutes stop consistency-check calls and return false
+        let timeOutCall = setTimeout(() => {
+          clearInterval(intervalCall);
+          callback(null, false);
+        }, 10 * 60 * 1000);
+      }
+    })
+    .catch(err => {
+      callback(err);
+    });
+
+  function checkConsistency(consistencyToken) {
+    reqOpts.consistencyToken = consistencyToken;
+
+    this.bigtable.request(
+      {
+        client: 'BigtableTableAdminClient',
+        method: 'checkConsistency',
+        reqOpts: reqOpts,
+      },
+      (err, resp) => {
+        return resp ? resp.consistent === 1 : false;
```

Comment:
This is an asynchronous function, so returning here will not work as expected. You will have to provide a callback to the method `checkConsistency`, and then execute the callback with this value, i.e.:

```js
function checkConsistency(consistencyToken, callback) {
  reqOpts.consistencyToken = consistencyToken;

  this.bigtable.request(
    {
      client: 'BigtableTableAdminClient',
      method: 'checkConsistency',
      reqOpts: reqOpts,
    },
    (err, resp) => {
      // Added error handling:
      if (err) {
        callback(err);
        return;
      }
      callback(null, resp.consistent === 1);
    }
  );
}
```

Reply:
Done